### PR TITLE
[NHentai] fix several bugs about search

### PR DESCRIPTION
**Pull request by ZIDOUZI**
1. remove `+` before language since they have different semantics (add `+` before means this is not concern as a namespace, that time `+` will be encoded to `%2B`).
2. change fallback value of `fixedQuery` to empty string rather than a quote, since they may have different semantics
3. change search url since there have a redirection.
4. add colon before tag. I forgotten in last pr, sorry about that.

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Comment by towermom9** [comment]
yo tested it and the search is now completely broken?

**Comment by towermom9** [comment]
apparently if I search using the "English" version it doesn't work

**Comment by towermom9** [comment]
but if I search with the "multi" one it works

**Comment by ZIDOUZI** [comment]
> yo tested it and the search is now completely broken?

before #4642 the language filter behave as `+language: specificLanguage`, that is not correct but work, I througt there is only one bug and fix it, now the remains bug destroyed the search

**Comment by ZIDOUZI** [comment]
> but if I search with the "multi" one it works

since the multi language not apply `+lanugage:xxx`, it will work.

**Comment by towermom9** [comment]
example:
Screenshot_2024-08-17-13-23-14-313_app mihon-edit
Screenshot_2024-08-17-13-23-41-459_app mihon-edit


Screenshot_2024-08-17-13-22-53-703_app mihon-edit
Screenshot_2024-08-17-13-24-24-135_app mihon

**Comment by towermom9** [comment]
> > yo tested it and the search is now completely broken?
> 
> before #4642 the language filter behavior as `+language: specificLanguage`, that is not correct but work, I througt there is only one bug and fix it, now the remains bug destroyed the search

oh Ic 😅

**Comment by AwkwardPeak7** [comment]
it seems you are not testing your prs... Please do test them

**Comment by ZIDOUZI** [comment]
> it seems you are not testing your prs... Please do test them

Already test. This will work well.

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt [hunk 1]
```diff
@@ -136,23 +136,23 @@ open class NHentai(
     }
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val fixedQuery = query.ifEmpty { "\"\"" }
+        val fixedQuery = query.ifEmpty { "" }
```
Comment (vetleledaal) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt]: `String.ifEmpty { "" }` doesn't make sense.

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**